keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -238,7 +239,7 @@ class SqlDeliveryConfigRepository(
       .and(ENVIRONMENT.DELIVERY_CONFIG_UID.eq(DELIVERY_CONFIG.UID))
       .fetchOne { (name) ->
         get(name)
-      }
+      } ?: throw OrphanedResourceException(resourceId)
```

Comment:
Not for this PR, but it looks like this exception won't trigger our flap detection veto and it would be nice to stop evaluating orphaned resources early on.